src/AppBundle/Entity/Repository/SurveyRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace AppBundle\Entity\Repository;
+
+use AppBundle\Entity\Department;
+use AppBundle\Entity\Semester;
+use AppBundle\Entity\Survey;
+use AppBundle\Entity\User;
+use Doctrine\ORM\EntityRepository;
+
+class SurveyRepository extends EntityRepository
+{
+    /**
+     * @param User $user
+     * @param Department $department
+     * @param Semester $semester
+     * @return Survey[]
+     */
+    public function findOneByUserNotTaken(User $user, Semester $semester)
+    {
+        $department = $user->getDepartment();
+
+        $qb = $this->_em->createQueryBuilder();
+
+        $exclude = $qb
+                ->select('IDENTITY(survey_taken.survey)')
+                ->from('AppBundle:SurveyTaken', 'survey_taken')
+                ->where('survey_taken.user = :user');
+
+        return $this->createQueryBuilder('survey')
+                ->select("survey")
+                ->where('survey.teamSurvey = true')
+                ->where('survey.semester =:semester')
+                ->andWhere('survey.department =:department OR survey.department is NULL')
```

Comment:
Hva tjener vi på å sjekke department her? Kan en bruker svare på surveys utenfor sin department? I såfall burde funksjonen hete `findOneNotTakenByUserInDepartmentAndSemester` eller lignende for å få med dette.